### ISSUE TITLE
Plugin Update error in DEV version

### ISSUE THREAD

**Issue by JackGruber** [laurent22/joplin#4723]
A plugin which is activated throws an error  during the update.
With deactivated plugins the update works fine.

I think the error cames with the update 044477ba0de35236e9c53d57792fa71ad61d7e31 from @mablin7 

## Environment

Joplin 1.8.0 (dev, win32)

Client ID: fbec70b4abf046b69b9d965d460cc5bc
Sync Version: 2
Profile Version: 34
Keychain Supported: Yes
Revision: 656673ed5 (dev)

## Steps to reproduce

1. Copy a outdated version of a plugin into  the plugin folder
2. Enable the plugin
3. Update the plugin on the GUI

## Logfile

![grafik](https://user-images.githubusercontent.com/24863925/112134770-57a63900-8bcd-11eb-968c-c857138d8c09.png)

```
Uncaught (in promise) TypeError: Cannot read property 'enabled' of undefined
    at C:\git\fork\joplin\packages\app-desktop\gui\ConfigScreen\controls\plugins\useOnInstallHandler.js:41
    at Immer.produce (C:\git\fork\joplin\packages\app-desktop\node_modules\immer\dist\immer.cjs.development.js:728)
    at Object.<anonymous> (C:\git\fork\joplin\packages\app-desktop\gui\ConfigScreen\controls\plugins\useOnInstallHandler.js:38)
    at Generator.next (<anonymous>)
    at fulfilled (C:\git\fork\joplin\packages\app-desktop\gui\ConfigScreen\controls\plugins\useOnInstallHandler.js:5)
```

![grafik](https://user-images.githubusercontent.com/24863925/112135993-b1f3c980-8bce-11eb-8d9c-1e704bdb71f8.png)

**Comment by mablin7** [comment]
Yes it looks like it's my fault. I'm looking into it right away.

**Comment by mablin7** [comment]
I can't reproduce it on 656673e 

![image](https://user-images.githubusercontent.com/7415668/112143494-3b5bc980-8bd8-11eb-9074-373ab58d3220.png)

I tried replicating everything as you described but I'm not getting any errors

**Comment by mablin7** [comment]
Do you only copy the jpl file, or something else too?

**Comment by JackGruber** [comment]
Hm I think this happens only when you install the plugin manually (coping into the plugings folder).
Because then the plugin is enabled, but no entry in the `settings` table `plugin.state` is written for this plugin.

So I think the `plugins.states` is only written when you install the plugin from the GUI or when you disable the plugin. 
All manually installed plugins have no entry in plugins.state `plugins.states`. 

![grafik](https://user-images.githubusercontent.com/24863925/112155872-05bddd00-8be6-11eb-9f5e-55bb0af45753.png)

**Comment by JackGruber** [comment]
Close Joplin, copy the jpl to the plugins folder. 
Yes this is supported.

**Comment by mablin7** [comment]
I'm sorry, I didn't know about this feature.
Then I guess a simple check before setting the field `enabled` in the hook should fix it.
